pkg/command/service/clean_test.go

```diff
@@ -0,0 +1,87 @@
+// Copyright © 2020 The Knative Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package service
+
+import (
+	"context"
+	"testing"
+
+	"gotest.tools/assert"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	k8sfake "k8s.io/client-go/kubernetes/fake"
+	clienttesting "k8s.io/client-go/testing"
+	"knative.dev/kperf/pkg"
+	"knative.dev/kperf/pkg/testutil"
+	servingv1client "knative.dev/serving/pkg/client/clientset/versioned/typed/serving/v1"
+	servingv1fake "knative.dev/serving/pkg/client/clientset/versioned/typed/serving/v1/fake"
+)
+
+func TestNewServiceCleanCommand(t *testing.T) {
+	t.Run("incompleted or wrong args for service clean", func(t *testing.T) {
+		client := k8sfake.NewSimpleClientset()
+		fakeServing := &servingv1fake.FakeServingV1{Fake: &clienttesting.Fake{}}
+		servingClient := func() (servingv1client.ServingV1Interface, error) {
+			return fakeServing, nil
+		}
+
+		p := &pkg.PerfParams{
+			ClientSet:        client,
+			NewServingClient: servingClient,
+		}
+		cmd := NewServiceCleanCommand(p)
+
+		_, err := testutil.ExecuteCommand(cmd)
+		assert.ErrorContains(t, err, "both namespace and namespace-prefix are empty")
+
+		_, err = testutil.ExecuteCommand(cmd, "--namespace-prefix", "test-kperf-", "--namespace-range", "2,1")
+		assert.ErrorContains(t, err, "failed to parse namespace range 2,1")
+	})
+
+	t.Run("clean service as expected", func(t *testing.T) {
+		client := k8sfake.NewSimpleClientset()
+		fakeServing := &servingv1fake.FakeServingV1{Fake: &clienttesting.Fake{}}
+		servingClient := func() (servingv1client.ServingV1Interface, error) {
+			return fakeServing, nil
+		}
+
+		p := &pkg.PerfParams{
+			ClientSet:        client,
+			NewServingClient: servingClient,
+		}
+
+		nsSpec1 := &v1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: "test-kperf-1"}}
+		_, err := p.ClientSet.CoreV1().Namespaces().Create(context.TODO(), nsSpec1, metav1.CreateOptions{})
```

Comment:
As namespace creation is end-user behavior, not included in kperf, so suggest to remove Namespaces().Creation(xxxx) here, and update to something like

```
		ns := &corev1.Namespace{
			ObjectMeta: metav1.ObjectMeta{
				Name:      "test-kperf-1"
			},
			Data: make(map[string]string),
		}
		client := k8sfake.NewSimpleClientset(ns)

```